sdk/purview/purview-sharing-rest/test/public/receivedSharesTest.spec.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  getLongRunningPoller,
+  InPlaceReceivedShareOutput,
+  OperationResponseOutput,
+  PurviewSharingClient,
+  ReceivedShareListOutput,
+} from "../../src";
+import { env, Recorder } from "@azure-tools/test-recorder";
+import { assert } from "chai";
+import { createClient, createRecorder } from "./utils/recordedClient";
+import { Context } from "mocha";
+
+describe("Received Shares Operations", () => {
+  let recorder: Recorder;
+  let client: PurviewSharingClient;
+
+  let receivedShareId = "";
+  const pollingIntervalMs = 30000;
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    client = createClient(recorder);
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("List all Detached Received Shares", async function () {
+    const response = await client.path("/receivedShares/detached").get({
+      queryParameters: { filter: "properties/StoreKind eq 'AdlsGen2Account'" },
+    });
+
+    assert.strictEqual(response.status, "200");
+    const detachedReceivedShares = response.body as ReceivedShareListOutput;
```

Comment:
Same comment as in the samples, please check for `isUnexpected` to get the right types